### PR TITLE
PLAT-1214: Forge Azure Container Registry

### PR DESCRIPTION
**Pull request by fraenkel**
Create a registry for Forge
    Create a managed user identity to push/pull to the Forge ACR
    Allow the AKS system identities to pull from ACR

    Since this is a module, test it like one.
    Move the providers to the caller.

### CODE REVIEW

**Review by steved — APPROVED** [review]

**Comment by fraenkel** [comment]
@steved Decided to remove all things unnecessary and simplify. Please verify.

File: forge.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+resource "azurerm_container_registry" "forge" {
+  name                = replace("${var.resource_group}forge", "/[^a-zA-Z0-9]/", "")
+  resource_group_name = data.azurerm_resource_group.aks.name
+  location            = data.azurerm_resource_group.aks.location
+  sku                 = var.registry_tier
+  admin_enabled       = false
+
+  // Premium only
+  public_network_access_enabled = var.registry_tier == "Premium" ? false : true
```
Comment (steved) [inline forge.tf:9]: Do we need to consider private endpoints or anything? (speed might be a factor given the massive base image)

**Review by steved — APPROVED** [review]
Might be worth a readme update with the new changes?

File: forge.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+resource "azurerm_container_registry" "forge" {
+  name                = replace("${var.resource_group}forge", "/[^a-zA-Z0-9]/", "")
```
Comment (steved) [inline forge.tf:2]: Not 100% sure exposing forge is the best over something generic, but there's an argument no customer will ever see it, so I'm not sure I care much either way.

File: forge.tf (hunk 1 above)
Comment (fraenkel) [reply forge.tf:9]: The issue is the tier. Its only supported with Premium hence the logic. If you try to enable it for Basic or Standard, TF will error out.
This is one of those things where cost comes into play.
Do we do one thing for dev and another for prod?

Speedwise I am not certain. Most of the clouds will optimize "public" traffic back to their networks anyhow. Its more whether the traffic every thinks about hitting the internet or not.

File: forge.tf (hunk 2 above)
Comment (fraenkel) [reply forge.tf:2]: I can switch it to some UUID or random characters. The name must be globally unique which is just annoying.

File: forge.tf (hunk 1 above)
Comment (steved) [reply forge.tf:9]: I think we can cross that bridge when we come to it. Given this is all generally used for dev, it's not of great concern immediately.

File: forge.tf (hunk 2 above)
Comment (fraenkel) [reply forge.tf]: switched to domino for now

**Pull request merged**